plugins/modules/katello_manifest.py

```diff
@@ -76,116 +61,71 @@
-                headers = {'content_type': 'multipart/form-data', 'multipart': 'true'}
-                if not module.check_mode:
-                    Subscription().upload(data=data, files=files, headers=headers)
-                changed = True
-        except IOError as e:
-            module.fail_json(msg="Unable to open the manifest file: %s" % e)
-        except TaskFailedError as e:
-            if "same as existing data" in e.args[0]:
-                pass
-            elif "older than existing data" in e.args[0]:
-                module.fail_json(msg="Manifest is older than existing data: %s" % e)
-            else:
-                module.fail_json(msg="Upload of the manifest failed: %s" % e)
-    elif state == 'absent' and manifest_present:
-        if not module.check_mode:
-            Subscription().delete_manifest(data={'organization_id': organization.id})
-        changed = True
-    elif state == 'refreshed':
-        if not manifest_present:
-            module.fail_json(msg="No manifest found to refresh.")
-        else:
-            if not module.check_mode:
-                Subscription().refresh_manifest(data={'organization_id': organization.id})
-            changed = True
-    return changed
+from ansible.module_utils.foreman_helper import KatelloEntityApypieAnsibleModule
 
 
 def main():
-    module = AnsibleModule(
+    module = KatelloEntityApypieAnsibleModule(
         argument_spec=dict(
-            server_url=dict(required=True),
-            username=dict(required=True, no_log=True),
-            password=dict(required=True, no_log=True),
-            verify_ssl=dict(type='bool', default=True),
-            organization=dict(required=True),
             manifest_path=dict(),
             state=dict(default='present', choices=['absent', 'present', 'refreshed']),
-            redhat_repository_url=dict(),
+            repository_url=dict(aliases=['redhat_repository_url']),
         ),
         required_if=[
             ['state', 'present', ['manifest_path']],
         ],
-        supports_check_mode=True,
     )
 
-    if has_import_error:
-        module.fail_json(msg=import_error_msg)
+    entity_dict = module.clean_params()
 
-    set_task_timeout(300000)  # 5 minutes
+    module.connect()
 
-    server_url = module.params['server_url']
-    username = module.params['username']
-    password = module.params['password']
-    verify_ssl = module.params['verify_ssl']
-    organization = module.params['organization']
-    manifest_path = module.params['manifest_path']
-    redhat_repository_url = module.params['redhat_repository_url']
-    state = module.params['state']
+    organization = module.find_resource_by_name('organizations', name=entity_dict['organization'], thin=False)
+    scope = {'organization_id': organization['id']}
 
-    create_server(server_url, (username, password), verify_ssl)
-    ping_server(module)
+    existing_manifest = organization.get('owner_details', {}).get('upstreamConsumer')
```

Comment:
`None` should have a dull `get`. ;)

Reply:
I see no `None.get` here, just a `{}.get` ;)

Reply:
Perhaps the longer version is more readable:
```suggestion
    try:
        existing_manifest = organization['owner_details']['upstreamConsumer']
    except KeyError:
        existing_manifest = None
```

Reply:
Sorry, i meant that as a feature request to python. ;)

Reply:
@mdellweg ah, maybe. But I like @ekohl's suggestion